kfs-core/src/main/java/edu/arizona/kfs/fp/document/DisbursementVoucherDocument.java

```diff
@@ -1,27 +1,205 @@
+import edu.arizona.kfs.fp.businessobject.PaymentMethod;
+import edu.arizona.kfs.fp.service.PaymentMethodGeneralLedgerPendingEntryService;
+import edu.arizona.kfs.sys.KFSConstants;
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+import edu.arizona.kfs.vnd.businessobject.VendorDetailExtension;
+import org.kuali.rice.krad.service.BusinessObjectService;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.rice.kns.service.DocumentHelperService;
+import org.kuali.rice.kns.document.authorization.DocumentAuthorizer;
+import org.kuali.rice.kew.api.KewApiConstants;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.kuali.rice.kns.document.authorization.TransactionalDocumentPresentationController;
+import org.kuali.rice.krad.util.KRADConstants;
+import org.kuali.rice.krad.util.ObjectUtils;
+import org.kuali.rice.coreservice.framework.parameter.ParameterConstants.COMPONENT;
+import org.kuali.rice.kew.framework.postprocessor.DocumentRouteStatusChange;
+
 
+/**
+ * Document class override to ensure that the bank code is synchronized with the
+ * payment method code.
+ */
+// This annotation is needed to make parameter lookups work properly
+@COMPONENT( component = "DisbursementVoucher" )
 public class DisbursementVoucherDocument extends org.kuali.kfs.fp.document.DisbursementVoucherDocument {
 
+    public static final String DOCUMENT_TYPE_DV_NON_CHECK = "DVNC";
+    
     private static final long serialVersionUID = 8820340507728738505L;
-
     private static Logger LOG = Logger.getLogger(DisbursementVoucherDocument.class);
+    private static PaymentMethodGeneralLedgerPendingEntryService paymentMethodGeneralLedgerPendingEntryService;
 
     @Override
     public void prepareForSave() {
         LOG.debug("DisbursementVoucherDocument.prepareForSave()");
         super.prepareForSave();
 
+        DocumentHelperService documentHelperService = SpringContext.getBean(DocumentHelperService.class);
+        DocumentAuthorizer docAuth = documentHelperService.getDocumentAuthorizer(this);
+
+        // First, only do this if the document is in initiated status - after that, we don't want to 
+        // accidentally reset the bank code
+        if ( KewApiConstants.ROUTE_HEADER_INITIATED_CD.equals( getDocumentHeader().getWorkflowDocument().getStatus().getCode() )
+                || KewApiConstants.ROUTE_HEADER_SAVED_CD.equals( getDocumentHeader().getWorkflowDocument().getStatus().getCode() ) ) {
+            // need to check whether the user has the permission to edit the bank code
+            // if so, don't synchronize since we can't tell whether the value coming in
+            // was entered by the user or not.        
+            if ( !docAuth.isAuthorizedByTemplate(this, 
+                    KFSConstants.ParameterNamespaces.KFS, 

```

Comment:
KFSConstants.ParameterNamespaces has been deprecated. Should use KFSConstants.CoreModuleNamespaces.